packages/amplication-server/src/core/auth/auth.service.ts

```diff
@@ -763,6 +774,59 @@ export class AuthService {
     return workspace as unknown as Workspace & { users: AuthUser[] };
   }
 
+  async loginOrSignUp(profile: AuthProfile, response: Response): Promise<void> {
+    let user = await this.getAuthUser({
+      account: {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        OR: [{ githubId: profile.sub }, { email: profile.email }],
+      },
+    });
+    let isNew: boolean;
+    const existingUser = !!user;
+    if (!user) {
+      user = await this.createUser(profile);
+      isNew = true;
+    }
+    if (!user.account.githubId || user.account.githubId !== profile.sub) {
+      user = await this.updateUser(user, { githubId: profile.sub });
+      isNew = false;
+    }
+
+    this.trackCompleteEmailSignup(
+      user.id,
```

Comment:
Pay attention here that the Anaytics tracking key should be the Account.id and not the User.id

Reply:
done